programs/drift/src/instructions/keeper.rs

```diff
@@ -426,6 +435,145 @@ pub fn handle_update_user_open_orders_count<'info>(ctx: Context<UpdateUserIdle>)
     Ok(())
 }
 
+pub fn place_swift_taker_order<'c: 'info, 'info>(
+    taker_key: Pubkey,
+    taker: &mut RefMut<User>,
+    taker_order_params_message: SwiftOrderParamsMessage,
+    ix_sysvar: &AccountInfo<'info>,
+    sig: [u8; 64],
+    perp_market_map: &PerpMarketMap,
+    spot_market_map: &SpotMarketMap,
+    oracle_map: &mut OracleMap,
+    state: &State,
+) -> Result<()> {
+    #[cfg(all(feature = "mainnet-beta", not(feature = "anchor-test")))]
+    {
+        panic!("Swift orders are disabled on mainnet-beta");
+    }
+
+    // Authenticate the signature
+    let ix_idx = load_current_index_checked(ix_sysvar)?;
+    validate!(
+        ix_idx > 0,
+        ErrorCode::InvalidVerificationIxIndex,
+        "instruction index must be greater than 0"
+    )?;
+    let ix: Instruction = load_instruction_at_checked(ix_idx as usize - 1, ix_sysvar)?;
+    verify_ed25519_ix(
+        &ix,
+        &taker.authority.to_bytes(),
+        &taker_order_params_message.clone().try_to_vec()?,
+        &sig,
+    )?;
+
+    // Verify inputs and validate market index
+    taker_order_params_message.verify_all_same_market_indexes()?;
+    let taker_order_params = taker_order_params_message.swift_order_params;
+    let clock = &Clock::get()?;
+
+    if taker_order_params.len() == 0 {
+        msg!("No orders to place");
+        return Err(print_error!(ErrorCode::SwiftOrderSequenceError)().into());
+    }
+
+    // First order must be a taker order
+    let matching_taker_order_params = taker_order_params[0];
+    if matching_taker_order_params.order_type != OrderType::Market {
+        msg!("First order must be a taker order");
+        return Err(print_error!(ErrorCode::SwiftOrderSequenceError)().into());
+    }
+    let expected_order_id = taker_order_params_message.expected_order_id;
+
+    let taker_next_order_id = taker.next_order_id;
+    let order_slot = taker_order_params_message.slot;
+    if expected_order_id.cast::<u32>()? == taker_next_order_id {
+        controller::orders::place_perp_order(
+            state,
+            taker,
+            taker_key,
+            perp_market_map,
+            spot_market_map,
+            oracle_map,
+            clock,
+            matching_taker_order_params,
+            PlaceOrderOptions {
```

Comment:
think it'd be nice to have a `swift_taker_orer_slot` method to set PlaceOrderOptions so it matches pattern elsewhere: https://github.com/drift-labs/protocol-v2/blob/master/programs/drift/src/state/order_params.rs#L709